lib/ansible/modules/notification/mail.py

```diff
@@ -241,9 +269,46 @@ def main():
     secure_state = False
     sender_phrase, sender_addr = parseaddr(sender)
 
+    pgp_recipients = module.params.get('pgp_recipients')
+    pgp = pgp_recipients or module.params.get('pgp')
+
     if not body:
         body = subject
 
+    if pgp:
+        try:
+            import gnupg
+        except ImportError:
+            module.fail_json(rc=1, msg='OpenPGP encryption requires python-gnupg')
+
+        # TODO: Implement proper PGP/MIME
+        if attach_files:
+            module.fail_json(rc=1, msg='OpenPGP encryption does not yet support attachments')
+
+        gpg = gnupg.GPG()
+        gpg.encoding = charset
+
+        if pgp_recipients:
+            import re
+            from ansible.module_utils.six import string_types
+
+            def is_invalid(s):
+                not (isinstance(s, string_types) and
+                     re.match(r'^0x([0-9A-F]+|[0-9a-f]+)$', s))
+
+            invalid_recipients = filter(is_invalid, pgp_recipients)
+            if invalid_recipients:
+                module.fail_json(rc=1, msg='pgp_recipients: Expected fingerprints, got ' + ', '.join(invalid_recipients))
+
+            encrypted = gpg.encrypt(body, pgp_recipients, always_trust=True)
+        else:
```

Comment:
Can not `recipients/to` and `gpg_recipients` parameters be used together ?

If they are exclusive, module documentation must be updated ([see](http://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-block))and `mutually_exclusive` ([example](https://github.com/ansible/ansible/blob/e5b3f60a7400dfd7de6d38724a0de5e3882f87f3/lib/ansible/modules/net_tools/basics/get_url.py#L399)) must be added in `agument_spec`.

Reply:
`to` and `gpg_recipients` can be used together, yes.
(The inline documentation contains an example)

Reply:
What bugs me is that either fingerprints or email addresses could be used in order to encrypt but not both. Why fingerprints and email addresses could not both be used ? Should not `gpg_recipients` be allowed to contain both fingerprint and email address (then `recipients/to` would not be used) ?

Reply:
Ah, I see.  I guess I could make `gpg_recipients` accept both, and only set `always_trust` when only fingerprints are used.

Mixing fingerprints and textual UIDs is pretty awkward, though: unlike fingerprints, UIDs can match several keys at once, and neither `python-gnupg` nor the `gpg` manpage seem to specify which key then gets selected.

Reply:
Then should not encryption using UIDs be removed ?

Reply:
I would be OK with that (I pin by fingerprint anyhow), but I suspect many would find it convenient to be able to “just encrypt stuff to the right recipient” (based on `to`), despite the looser/fuzzier guarantees.

Reply:
So, should we remove the support for UIDs entirely?

Reply:
Using `to` when `gpg_recipients` isn't set seems confusing. Compared to the current behavior. I would prefer:
- either mix UIDs and fingerprints in `gpg_recipients`
- or remove the support for UIDs
@mscherer any advice ?